File: OpenCV/07_gradientes_bordes/python/main.py
import cv2
img=cv2.imread("input.jpg",cv2.IMREAD_GRAYSCALE); assert img is not None
gx=cv2.Sobel(img,cv2.CV_32F,1,0,ksize=3); gy=cv2.Sobel(img,cv2.CV_32F,0,1,ksize=3); canny=cv2.Canny(img,80,160)
cv2.imwrite("output_canny.png",canny)

File: OpenCV/07_gradientes_bordes/cpp/main.cpp
#include <opencv2/opencv.hpp>
int main(){ auto g=cv::imread("input.jpg",cv::IMREAD_GRAYSCALE); if(g.empty()) return 1; cv::Mat gx,gy,e; cv::Sobel(g,gx,CV_32F,1,0); cv::Sobel(g,gy,CV_32F,0,1); cv::Canny(g,e,80,160); cv::imwrite("output_canny.png",e); }
